### PR TITLE
fix: correct SVE accelerator bugs in shufti, truffle, and noodle

### PR DESCRIPTION
**Pull request by byeonguk-jeong**
## Summary

Fix three bugs affecting SVE/SVE2 accelerator code paths, with regression tests for truffle.

### shufti-double: false positive on vector boundary
Use predicate mask to prevent false positives when scanning double shufti across vector boundaries.

### truffle: off-by-one in rtruffleExecSVE tail
Fix off-by-one error in reverse truffle SVE tail processing. Includes comprehensive unit tests covering forward/reverse edge cases, varying lengths, alignment sweep, high-byte characters, and NUL matching.

### noodle SVE: incorrect return types and scan length
- Change return type from `hwlmcb_rv_t` to `hwlm_error_t` to match actual function signatures.
- Fix `scanDouble` short-path condition to use `(e - d)` instead of `scan_len`, which could be stale after adjusting `d` for history.
- Fix signedness mismatch in the pointer distance comparison.
- Fix formatting: add space after `if` keyword.

## Commits

- f58f253227b8 shufti-double: use predicate mask to prevent false positives
- 6d8becc2bf28 truffle: fix off-by-one in rtruffleExecSVE tail and add unit tests
- 2414ae9e2004 hwlm: correct return types and scan length in SVE noodle engine

## Files Changed

- src/nfa/shufti_sve.hpp
- src/nfa/truffle_simd.hpp
- src/hwlm/noodle_engine_sve.hpp
- unit/internal/truffle.cpp
- unit/internal/shufti.cpp

@AhnLab-OSS @AhnLab-OSSG

### CODE REVIEW

File: src/hwlm/noodle_engine_sve.hpp [hunk 1]
```diff
@@ -244,15 +244,15 @@ hwlm_error_t scanDouble(const struct noodTable *n, const u8 *buf, size_t len,
 
     svuint8_t chars = svreinterpret_u8(getCharMaskDouble(n->key0, n->key1, noCase));
 
-    if (scan_len <= svcntb()) {
+    if ((size_t)(e - d) <= svcntb()) {
```
Comment (ypicchi-arm) [inline src/hwlm/noodle_engine_sve.hpp:247]: Why this? scan_len is defined as (e - d) already just a few lines above.

File: src/nfa/shufti_sve.hpp [hunk 2]
```diff
@@ -173,7 +173,7 @@ svbool_t doubleMatched(svuint8_t mask1_lo, svuint8_t mask1_hi,
     svuint8_t t      = svorr_x(svptrue_b8(), merged_t1, t2);
     *inout_t1 = new_t1;
 
-    return svnot_z(svptrue_b8(), svcmpeq(svptrue_b8(), t, (uint8_t)0xff));
+    return svnot_z(pg, svcmpeq(pg, t, (uint8_t)0xff));
```
Comment (ypicchi-arm) [inline src/nfa/shufti_sve.hpp]: I agree for the svnot_z. Somehow I used _z instead of _x, but did not use the predicate.
I believe we can keep the svptrue for the compare though, for the cases where full vectors are faster than partial ones.

File: src/hwlm/noodle_engine_sve.hpp (hunk 1 above)
Comment (byeonguk-jeong) [reply src/hwlm/noodle_engine_sve.hpp:247]: As you can see at line 243, d has incremented.

File: src/nfa/shufti_sve.hpp (hunk 2 above)
Comment (byeonguk-jeong) [reply src/nfa/shufti_sve.hpp]:
```suggestion
    return svnot_x(pg, svcmpeq(svptrue_b8(), t, (uint8_t)0xff));
```
That makes sense. How about this?

Comment (ypicchi-arm) [reply src/nfa/shufti_sve.hpp]: Oh, sorry, I didn't mean to change _z to _x. _x mean it's left in an undefined state, which is similar to what happen when we do _z with a full predicate but the tail is garbage. So that's wrong.
One should use _z so that the non predicated lines are explicitly set to 0

```suggestion
    return svnot_z(pg, svcmpeq(svptrue_b8(), t, (uint8_t)0xff));
```

Comment (byeonguk-jeong) [reply src/nfa/shufti_sve.hpp]: I was mistaken. I thought _x makes the inactive lanes as-is. Thank you for the correction.

File: src/hwlm/noodle_engine_sve.hpp (hunk 1 above)
Comment (ypicchi-arm) [reply src/hwlm/noodle_engine_sve.hpp:247]: Oh, I missed that one. I don't remember why we increment d here, but you change make sense.

**Comment by markos** [comment]
Again I will ask you to rebase against current develop branch.

**Comment by byeonguk-jeong** [comment]
> Again I will ask you to rebase against current develop branch.

Done :)

**Review by markos — APPROVED** [review]
The remaining tests failed for different reasons. Looks good, thank you for your contribution!

**Review by markos — APPROVED** [review]
The remaining tests failed for different reasons. Looks good, thank you for your contribution!
